app/playground/page.tsx

```diff
@@ -138,6 +153,10 @@ export default function PlaygroundPage() {
 
   const handleParameterChange = (key: keyof Parameters, value: number | boolean | string) => {
     setParameters(prev => ({ ...prev, [key]: value }));
+
+    if (key !== "apiKey") {
+      updatePlaygroundSettings({ [key]: value });
```

Comment:
you should call `updatePlaygroundSettings` only when save button is pressed, because it's saving it to localStorage there.